src/FeatureExtractor/Mobilenet.js

```diff
@@ -40,12 +40,8 @@ class Mobilenet {
     this.isPredicting = false;
     this.mapStringToIndex = [];
     this.usageType = null;
-
-    this.loadModel().then((net) => {
-      this.modelLoaded = true;
-      this.mobilenetFeatures = net;
-      callback();
-    });
+    this.ready = callCallback(this.loadModel(), callback);
+    this.then = this.ready.then;
```

Comment:
I don't know if this is to spec, as I don't think the `this` is bound by default on a promise. Perhaps `this.ten = this.ready.then.bind(this.ready)`

I am also not sure about using `ready` as the name of the Promise, it seems to me that ready implies a boolean as it is in other places in the library. Perhaps `readyPromise` or `loadingPromise`? I'm not sure but `await model.ready` doesn't read very well to me...

Reply:
I'm fine with any name for this, but note that this PR changes all `ready` properties to be Promises, so there is no boolean implication. I don't think a `Promise` suffix is appropriate; that isn't the pattern anywhere else in JavaScript, like the Response object of a fetch call does not have `.textPromise`, it has `.text()`.

Also, `.ready` is an completely internal property: it isn't super worthwhile to bikeshed the name.

Reply:
Is it internal though? There's at least one [example using it](https://github.com/ml5js/ml5-examples/blob/9abe6b284c81f80b05f44f828a3a5b551a62ecff/p5js/StyleTransfer_Image/sketch.js#L36). I'm just not a fan of changing exposed APIs that aren't explicitly private. I would possibly be fine with just `_ready` or `readyInternal` such that the private nature of the property is clear. This is something that should be decided for the whole library so that internal properties don't appear to be public. I would personally push for prefixing private properties with `_`, but that's because I use the convention from Python.

Reply:
That example uses the `master` branch. I defer to the maintainers on the rest of this PR.